cfme/middleware/deployment.py

```diff
@@ -0,0 +1,127 @@
+from cfme.common import Validatable
+from cfme.fixtures import pytest_selenium as sel
+from cfme.web_ui import listaccordion as list_acc
+from cfme.web_ui.menu import nav, toolbar as tb
+from mgmtsystem.hawkular import Deployment, Path
+from utils.db import cfmedb
+from utils.varmeth import variable
+from . import list_tbl, MiddlewareBase
+
+
+def _db_select_query(name=None, server=None):
+    t_md = cfmedb()['middleware_deployments']
+    query = cfmedb().session.query(t_md.nativeid, t_md.name, t_md.ems_ref)
+    if name:
+        query = query.filter(t_md.name == name)
+    if server:
+        query = query.filter(t_md.nativeid.like('%{}%'.format(server)))
+    return query
+
+
+def _nav_through_provider(ctx):
+    if 'server' in ctx:
+        sel.force_navigate('middleware_server_detail',
+                           context={'provider': ctx['provider'], 'name': ctx['server']})
+    else:
+        sel.force_navigate('middleware_provider_detail', context=ctx)
+    list_acc.select('Relationships', 'Middleware Deployments', by_title=False, partial=True)
+
+
+def _nav_to_deployment(ctx):
+    if 'provider' in ctx:
+        _nav_through_provider(ctx)
+    if 'name' in ctx:
+        list_tbl().click_row_by_cells({'Deployment Name': ctx['name']})
+
+
+def _select_deployment(ctx):
+    if 'provider' in ctx:
+        _nav_through_provider(ctx)
+    if 'name' in ctx:
+        list_tbl().select_row('Deployment Name', ctx['name'])
+
+
+nav.add_branch(
+    'middleware_deployments', {
+        'middleware_deployment': lambda ctx: _select_deployment(ctx),
+        'middleware_deployment_detail': lambda ctx: _nav_to_deployment(ctx)
+    }
+)
+
+
+class MiddlewareDeployment(MiddlewareBase, Validatable):
+    property_tuples = [('name', 'name')]
+
+    def __init__(self, provider=None, **kwargs):
+        self.id = kwargs['id'] if 'id' in kwargs else None
+        self.name = kwargs['name'] if 'name' in kwargs else None
+        self.server = kwargs['server'] if 'server' in kwargs else None
+        self.provider = provider
+
+    def load_details(self, refresh=False, detail=True):
+        ctx = {}
+        if self.server:
+            ctx.update({'server': self.server})
+        if self.provider:
+            ctx.update({'provider': self.provider})
+        if self.name:
+            ctx.update({'name': self.name})
+        if detail is True:
+            ctx.update({'detail': True})
+            if not self.name:
+                raise KeyError('Variable "name" is not set!')
```

Comment:
This is handled when the context dictionary is created.

If you look in theconstructor, this branch can never happen because the name is set always.